http/http-advanced/src/main/java/io/quarkus/ts/http/advanced/GrpcServerInterceptors.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.ts.http.advanced;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Produces;
+
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.ServerInterceptor;
+import io.quarkus.grpc.GlobalInterceptor;
+
+class GrpcServerInterceptors {
+    @GlobalInterceptor
+    @ApplicationScoped
+    static class TypeTarget extends Base {
+    }
+
+    static class MethodTarget extends Base {
+    }
+
+    static class Producer {
+        @GlobalInterceptor
+        @Produces
+        MethodTarget methodTarget() {
+            return new MethodTarget();
+        }
+    }
+
+    abstract static class Base implements ServerInterceptor {
+        @Override
+        public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<ReqT, RespT> call, Metadata metadata,
+                ServerCallHandler<ReqT, RespT> next) {
+            GrpcResource.invoked.add(getClass().getName());
```

Comment:
check it's duplicated context https://quarkus.io/guides/duplicated-context#distinguish-between-root-and-duplicated-contexts (illegal state if it isn't) and put it to local data. Then access it in your endpoint and put it to response

```
io.quarkus.vertx.core.runtime.context.VertxContextSafetyToggle#validateContextIfExists
io.vertx.core.Context#putLocal
```

Note: don't know gRPC that well, you can maybe go with metadata instead? your choice

Reply:
```
            VertxContextSafetyToggle.validateContextIfExists("", "");
            final var ctx = Vertx.currentContext();
            ctx.putLocal(TokenCredential.class.getName(), token);
```